contracts/contracts/strategies/sonic/SonicSwapXAMOStrategy.sol

```diff
@@ -0,0 +1,710 @@
+    }
+
+    function _deposit(address _wS, uint256 _wsAmount) internal {
+        require(_wsAmount > 0, "Must deposit something");
+        require(_wS == ws, "Unsupported asset");
+
+        // Calculate the required amount of OS to mint based on the wS amount.
+        uint256 osDepositAmount = _calcTokensToMint(_wsAmount);
+
+        // Mint the required OS tokens to this strategy
+        IVault(vaultAddress).mintForStrategy(osDepositAmount);
+
+        // Add wS and OS liquidity to the pool and stake in gauge
+        _depositToPoolAndGauge(_wsAmount, osDepositAmount);
+
+        // Ensure solvency of the vault
+        _solvencyAssert();
+
+        // Emit event for the deposited wS tokens
+        emit Deposit(_wS, pool, _wsAmount);
+        // Emit event for the minted OS tokens
+        emit Deposit(os, pool, osDepositAmount);
+    }
+
+    /**
+     * @notice Deposit the strategy's entire balance of Wrapped S (wS) into the pool
+     */
+    function depositAll() external override onlyVault nonReentrant skimPool {
+        uint256 balance = IERC20(ws).balanceOf(address(this));
+        if (balance > 0) {
+            _deposit(ws, balance);
+        }
+    }
+
+    /***************************************
+                    Withdraw
+    ****************************************/
+
+    /**
+     * @notice Withdraw wS and OS from the SwapX pool, burn the OS,
+     * and transfer the wS to the recipient.
+     * @param _recipient Address to receive withdrawn asset which is normally the Vault.
+     * @param _asset Address of the Wrapped S (wS) contract.
+     * @param _wsAmount Amount of Wrapped S (wS) to withdraw.
+     */
+    function withdraw(
+        address _recipient,
+        address _asset,
+        uint256 _wsAmount
+    ) external override onlyVault nonReentrant skimPool {
+        require(_wsAmount > 0, "Must withdraw something");
+        require(_asset == ws, "Unsupported asset");
+
+        // Calculate how much pool LP tokens to burn to get the required amount of wS tokens back
+        uint256 lpTokens = _calcTokensToBurn(_wsAmount);
+
+        // Withdraw pool LP tokens from the gauge and remove assets from from the pool
+        _withdrawFromGaugeAndPool(lpTokens);
+
+        // Burn all the removed OS and any that was left in the strategy
+        uint256 osToBurn = IERC20(os).balanceOf(address(this));
+        IVault(vaultAddress).burnForStrategy(osToBurn);
+
+        // Transfer wS to the recipient
+        // Note there can be a dust amount of wS left in the strategy as
+        // the burn of the pool's LP tokens is rounded up
+        require(
+            IERC20(ws).balanceOf(address(this)) >= _wsAmount,
+            "Not enough wS removed from pool"
+        );
+        IERC20(ws).safeTransfer(_recipient, _wsAmount);
+
+        // Ensure solvency of the vault
+        _solvencyAssert();
+
+        // Emit event for the withdrawn wS tokens
+        emit Withdrawal(_asset, pool, _wsAmount);
```

Comment:
super nit: could be `_ws` instead of `_asset` for tiny code readability improvement

Reply:
I agree. Fixed